cmod/conditions.py

```diff
@@ -0,0 +1,160 @@
+        file and the required format and try again.""")
+      # TODO: might want to add logic that if the required conditions are not provided, run the process to calculate those that are missing or even all of them as don't want to trust an "incomplete" set of conditions
+      return False
+
+  # saves gantry conditions to a file
+  def save_gantry_conditions(self, filename=""):
+    if filename != "":
+      self.gantry_conditions_filename = filename
+    if self.gantry_conditions_filename == "":
+      self.create_gantry_conditions_filename()
+      self.increment_use_count()
+
+    with open(self.gantry_conditions_filename, 'w') as f:
+      f.write(json.dumps(self.gantry_conditions, indent=2))
+  
+  # returns the gantry conditions
+  def get_gantry_conditions(self):
+    return self.gantry_conditions
+  
+  def update_gantry_and_sipm_conditions(self, cmd, detid, z):
+    if cmd == 'visualcenterdet':
+      if cmd.board.lumi_coord_hasz(detid, z):
+        h = cmd.board.get_lumi_coord(detid, z)-cmd.board.get_vis_coord(detid, z)
+        cmd.board.add_lumi_vis_separation(detid, z, h)
+        # check if we have multiple H values out of tolerance with each other,
+        if self.is_h_valid(self.h_list, h, 0.5):
+          self.h_list.append(h)
+          self.gantry_conditions["lumi_vs_FOV_center"]["separation"] = ((self.gantry_conditions["lumi_vs_FOV_center"]['data']["separation"]*len(self.h_list)) + h) /  (len(self.h_list)+1)
+        # TODO: add the else: an error should be raised such that the operator knows that something is wrong (maybe the gantry head dislodged or was tugged
+    elif cmd == 'halign':
+      if cmd.board.vis_coord_hasz(detid, z):
+        h = cmd.board.get_lumi_coord(detid, z)-cmd.board.get_vis_coord(detid, z)
+        cmd.board.add_lumi_vis_separation(detid, z, h)
+
+        # check if we have multiple H values out of tolerance with each other,
+        if self.is_h_valid(self.h_list, h, 0.5):
+          self.h_list.append(h)
+          self.gantry_conditions["lumi_vs_FOV_center"]["separation"] = ((self.gantry_conditions["lumi_vs_FOV_center"]["separation"]*len(self.h_list)) + h) /  (len(self.h_list)+1)
+        # TODO: add the else: an error should be raised such that the operator knows that something is wrong (maybe the gantry head dislodged or was tugged
+    elif cmd == 'visualhscan':
+      visM = cmd.board.get_visM(id, 5)
+      self.gantry_conditions['FOV_to_gantry_coordinates']['z'] = visM['z']
+      self.gantry_conditions['FOV_to_gantry_coordinates']['transform'] = visM['data']['transform']
+
+    self.save_gantry_conditions()
+
+  def is_h_valid(self, h_list, h, tolerance):
+    """
+    Checks if the h value is within tolerance of the h values in the h_list
+    """
+    for h_i in h_list:
+      if abs(h_i-h) > tolerance:
+        return False
+    return True
+
+  # increments the use count
+  def increment_use_count(self):
+    self.gantry_conditions.gc_use_count += 1
```

Comment:
`self.gantry_conditions_use_count += 1`?

Reply:
To keep track of how many times the current gantry conditions file was used. Although, actually we need to further discuss how we want this to happen... where you mentioned that we would recalculate gantry conditions after x number of tileboards or something of that sort. Let's chat on this.